src/ml/neural_net/style_transfer/mps_style_transfer.m

```diff
@@ -0,0 +1,419 @@
+      = [MPSNNAdditionNode nodeWithSources:@[styleLossNode1.resultImage,
+                                             styleLossNode2.resultImage]];
+
+    MPSNNAdditionNode* addLossStyle3Style4
+      = [MPSNNAdditionNode nodeWithSources:@[styleLossNode3.resultImage,
+                                             styleLossNode4.resultImage]];
+
+    MPSNNAdditionNode* addTotalStyleLoss 
+      = [MPSNNAdditionNode nodeWithSources:@[addLossStyle1Style2.resultImage,
+                                             addLossStyle3Style4.resultImage]];
+
+    MPSNNAdditionNode* totalLoss
+      = [MPSNNAdditionNode nodeWithSources:@[addTotalStyleLoss.resultImage,
+                                             contentLossNode.resultImage]];
+
+    totalLoss.resultImage.handle = [[TCMPSGraphNodeHandle alloc] initWithLabel:@"totalLossValue"];
+    totalLoss.resultImage.exportFromGraph = YES;
+    totalLoss.resultImage.synchronizeResource = YES;
+    totalLoss.resultImage.imageAllocator = [MPSImage defaultAllocator];
+
+    MPSNNInitialGradientNode *initialGradient
+      = [MPSNNInitialGradientNode nodeWithSource:totalLoss.resultImage];
+
+    BOOL resultsAreNeeded[] = { YES, YES };
+
+    NSArray<MPSNNFilterNode*>* lastNodes 
+      = [initialGradient trainingGraphWithSourceGradient:initialGradient.resultImage
+                                             nodeHandler: nil];
+
+    _trainingGraph = [MPSNNGraph graphWithDevice:_dev
+                                    resultImages:@[lastNodes[0].resultImage, lastNodes[1].resultImage]
+                                resultsAreNeeded:&resultsAreNeeded[0]];
+
+    _inferenceGraph = [MPSNNGraph graphWithDevice:_dev
+                                      resultImage:_model.forwardPass
+                              resultImageIsNeeded:YES];
+
+    _trainingGraph.format = MPSImageFeatureChannelFormatFloat32;
+    _inferenceGraph.format = MPSImageFeatureChannelFormatFloat32;
+  }
+  return self;
+}
+ 
+- (NSDictionary<NSString *, NSData *> *)exportWeights {
+  return [_model exportWeights];
+}
+
+- (NSDictionary<NSString *, NSData *> *)predict:(NSDictionary<NSString *, NSData *> *)inputs {
+  NSUInteger contentWidth;
+  NSUInteger contentHeight;
+  NSUInteger contentChannels;
+
+  [inputs[@"imageWidth"] getBytes:&contentWidth length:sizeof(contentWidth)];
+  [inputs[@"imageHeight"] getBytes:&contentHeight length:sizeof(contentHeight)];
+  [inputs[@"imageChannels"] getBytes:&contentChannels length:sizeof(contentChannels)];
+
+  MPSImageDescriptor *imgDesc = [MPSImageDescriptor
+    imageDescriptorWithChannelFormat:MPSImageFeatureChannelFormatFloat32
+                               width:contentWidth
+                              height:contentHeight
+                     featureChannels:contentChannels
+                      numberOfImages:1
+                               usage:MTLTextureUsageShaderWrite | MTLTextureUsageShaderRead];
+
+  NSMutableArray<MPSImage *> *contentImageArray;
```

Comment:
Another case where you should instantiate an object.

In Obj-C code you should basically never have code that looks like
```
NSObject *foo;
```
unless it's part of something like
```
NSObject *foo;
if (...) {
  foo = ...;
} else {
  foo = ...;
}
```

Reply:
Fixed this!